src/auth/basic.rs

```diff
@@ -57,7 +57,7 @@ impl Auth for BasicAuth {
 
         let expiration: u64 = map
             .get("expires_in")
-            .ok_or_else(|| "No `expires_in` field in response")?
+            .ok_or_else(|| "Authentication error: No `expires_in` field in response")?
```

Comment:
While this makes the error a little more clear, a more proper and long-term solution would be to implement an ApiError struct that the response would deserialize into. Only if you’re interested in doing that though.